libs/domain/checkout/place-order/place-order.component.ts

```diff
@@ -0,0 +1,32 @@
+import { CheckoutMixin } from '@spryker-oryx/checkout';
+import { resolve } from '@spryker-oryx/di';
+import { RouterService } from '@spryker-oryx/router';
+import { hydratable, i18n } from '@spryker-oryx/utilities';
+import { html, LitElement, TemplateResult } from 'lit';
+
+@hydratable('window:load')
+export class CheckoutPlaceOrderComponent extends CheckoutMixin(LitElement) {
+  protected router = resolve(RouterService);
+
+  protected override render(): TemplateResult | void {
+    if (!this.isAvailable()) return;
+
+    return html`<oryx-button
+      ?inert=${this.isBusy()}
+      ?loading=${this.isBusy()}
+      @click="${this.onClick}"
+    >
+      <button>${i18n('checkout.place-order')}</button>
+    </oryx-button>`;
+  }
+
+  protected onClick(): void {
+    this.checkoutService.placeOrder().subscribe((response) => {
+      this.redirect(response.redirectUrl, response.isExternalRedirect);
+    });
+  }
+
+  protected redirect(url?: string, external = false): void {
```

Comment:
Why do we need second params? Also not sure that this is ok to have functions with optional param. Mb better change url to required prop and add condition when we call this function?